### DISCUSSION TITLE
Question on how to make my own Local Library

### DISCUSSION THREAD

**Discussion by wwookE** [visose/Robots#180] [Q&A] [answered]
<img width="1280" alt="Screenshot 2022-09-05 150034" src="https://user-images.githubusercontent.com/112847077/188372043-756f99c9-7639-41f9-a67b-2250426414c5.png">

Hi! I have question on how to move uem-IRB2600 to 0,0,0 xy plane and save it on my local library.

<img width="841" alt="Screenshot 2022-09-05 150205" src="https://user-images.githubusercontent.com/112847077/188372448-44e98bcd-03d3-4381-89d4-37a33d3ac2bc.png">
<img width="1280" alt="Screenshot 2022-09-05 150553" src="https://user-images.githubusercontent.com/112847077/188372521-3a1810a1-b5a8-43d3-aa05-9eb7eb506f4b.png">
<img width="787" alt="Screenshot 2022-09-05 151439" src="https://user-images.githubusercontent.com/112847077/188372578-939d2972-2866-4f48-930d-5ff9919a65d2.png">

I've downloaded uem.3dm and uem.xml from github, placed them on 'C:\Users\desig(myusername)\Documents\Robots' and modified the 'Base' script of xml file to make the robot set on 0,0,0 

<img width="856" alt="Screenshot 2022-09-05 150718" src="https://user-images.githubusercontent.com/112847077/188372844-6055089d-9d8f-4bf9-bd20-238c23a82fb0.png">

Then, I opened grasshopper and used LoadRobotSystem component to find my local library, but I can't find it. 

Is there any way I can load my local library to LoadRobotSystem?

Thanks :)

**Comment by visose** [comment]
That's strange, it seems correct. Can you share the content of the XML file? Not a screenshot from Internet Explorer, since it reformats it, but the actual text.

**Reply by wwookE** [reply]
[ABB2600 - Copy.txt](https://github.com/visose/Robots/files/9488277/ABB2600.-.Copy.txt)

Thank you for the reply! This is the content of the xml file

**Reply by visose** [reply]
From what you posted above, you should be seeing a library named "ABB2600" in the list (I mean in the window titled "Robot Libraries", not the dropdown widget connected to the component). It might appear more at the top, in your screenshot you have the list scrolled to the bottom.

If it does appear, then the reason you cannot load it is because it has the same name as the UEM one, and the component will load the first robot cell that finds with that name. Try changing the name of the robotcell:
from `<RobotCell name="uem-IRB2600" manufacturer="ABB">` to
`<RobotCell name="myOrg-IRB2600" manufacturer="ABB">`

**Reply by wwookE** [reply]
Thanks again! But I don't seem to understand what it means.....

"From what you posted above, you should be seeing a library named "ABB2600" in the list (I mean in the window titled "Robot Libraries", not the dropdown widget connected to the component). It might appear more at the top, in your screenshot you have the list scrolled to the bottom."

Are you saying that there is a folder named 'Robot Libraries' in the window and a library named 'ABB2600' will be in there...?

**Reply by visose** [reply]
I meant here: 
<img width="683" alt="image" src="https://user-images.githubusercontent.com/5129614/188590549-ce1b3f11-eb38-4295-8138-6d4848c64615.png">
Scroll up and check if ABB2600 is there. Also change the name of the robotcell.

**Reply by wwookE** [reply]
In the end, I solved this but it seems I also have to make local library (I mean the 'Robots' folder) at different location.

At first, I changed the name of robotcell as you mentioned, but couldn't find the robot from 'LoadRobotSystem' component in grasshopper. Then I made new 'Robots' folder in different location and it worked! 

<img width="846" alt="Screenshot 2022-09-07 101544" src="https://user-images.githubusercontent.com/112847077/188767664-d45e57ac-f38f-4d49-8792-0ad934de2714.png">

It was the first location that I made my 'Robots' folder.

<img width="842" alt="Screenshot 2022-09-07 101503" src="https://user-images.githubusercontent.com/112847077/188768139-be5879fe-a556-48c6-8133-0fca8e300e63.png">

Then I made new 'Robots' folder at this location and it worked!

<img width="1280" alt="Screenshot 2022-09-07 102940" src="https://user-images.githubusercontent.com/112847077/188768818-d779cdff-b10c-465b-8221-4ff7288738eb.png">
This it the image that I loaded my local Robot library in grasshopper

Again, I really appreciate all your replies : )

**Reply by visose** [reply]
My guess is you have multiple user accounts on the same computer, and you saved it on the folder for a different one than the one you're logged in.

**Comment by visose** [comment, marked as answer]
Closing this as resolved. The local library needs a valid `.xml`/`.3dm` pair in the Robots library folder for the Windows user actually running Rhino, and the `RobotCell name` should be unique if you are copying an existing library. In this case the working fix was creating the library in the correct user/location and avoiding a duplicate cell name.
